app/filters/good_job/base_filter.rb

```diff
@@ -31,6 +31,15 @@ def queues
                 .to_h
     end
 
+    def labels
+      filtered_query(params.slice(:queue_name))
+        .joins("CROSS JOIN unnest(labels) AS label")
+        .group("label")
+        .order("label")
+        .count
+        .to_h
```

Comment:
As you mentioned, this probably needs to be moved into something that can be done asynchronously